crates/chain/src/tx_graph.rs

```diff
@@ -755,13 +747,18 @@ impl<A: Clone + Ord> TxGraph<A> {
             }
         }
 
-        changeset.anchors = update.anchors.difference(&self.anchors).cloned().collect();
+        changeset.anchors = update
+            .anchors
+            .iter()
+            .filter(|(k, _)| !self.anchors.contains_key(k))
+            .map(|(k, v)| (*k, v.clone()))
+            .collect::<BTreeMap<_, _>>();
```

Comment:
We also need to compare values to see if things are changed. This means values `A` (anchor meta) needs to impl `PartialEq` for this.

This also needs a test.